### PR TITLE
Check IAM rules before handling requests

### PR DESCRIPTION
**Pull request by fvennetier**
IAM rules are loaded from a file by the "iam" middleware. Then any request controller can call this middleware with the help of the `check_iam_access` decorator, which will raise `AccessDenied` in case the operation is not allowed for the authenticated user.

Change-Id: I04739d7e44497bb25f64c60c0a6a2f77fc2dfc0b
Co-authored-by: Michael Bonfils <michael.bonfils@ovhcloud.com>

### CODE REVIEW

File: etc/s3-iam.cfg.in [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+connection = file://%RULES_FILE%
+
+[filter:s3api]
+use = egg:swift#s3api
+location = RegionOne
+force_swift_request_proxy_log = true
+log_level = DEBUG
+storage_domain = localhost
+s3_acl = true
+# As of October 2020, cannot mix IAM and ACLs
```
Comment (murlock) [inline etc/s3-iam.cfg.in]: Is it still true with iam.py l.329 ?

**Review by murlock — APPROVED** [review]
LGM
However it seems that tests covering `swift.acl.explicit_allow` header missing ?

File: etc/s3-iam.cfg.in (hunk 1 above)
Comment (fvennetier) [reply etc/s3-iam.cfg.in]: I removed the comment. `s3_acl_openbar` is not required any more (it is even not implemented).

**Comment by fvennetier** [comment]
> However it seems that tests covering `swift.acl.explicit_allow` header missing ?

ACLs are always placed on buckets and objects. Legacy tests suites required the `s3_acl_openbar = true` parameters to ignore ACLs in order to run. This parameter does not exist any more, but the test suites run fine. :slightly_smiling_face: 

Notice that anonymous requests still do not work. In that case IAM rules are checked **before** ACLs, and thus deny the request.

**Comment by fvennetier** [comment]
Ok, there is now a test for "public-read" ACLs and anonymous access.

**Review by AymericDu — APPROVED** [review]

**Pull request merged**